h5netcdf/core.py

```diff
@@ -691,15 +724,25 @@ def __init__(self, parent, name):
 
         self._dimensions = Dimensions(self)
         self._enumtypes = _LazyObjectLookup(self, self._enumtype_cls)
+        self._vltypes = _LazyObjectLookup(self, self._vltype_cls)
+        self._cmptypes = _LazyObjectLookup(self, self._cmptype_cls)
+        self._usertypes = dict()
```

Comment:
Unfortunately we can't utilize _LazyObjectLookup in a neat manner for a joined usertype-dict. Using a simple dict for now, just adding name/h5name key/value-pairs.

Reply:
Fixed!